pkg/reconciler/dependencybuild/buildrecipeyaml.go

```diff
@@ -77,8 +77,7 @@ func createPipelineSpec(maven bool, namespace string, commitTime int64, userConf
 			{Name: PipelineEnforceVersion, Type: pipelinev1beta1.ParamTypeString},
 			{Name: PipelineRequestProcessorImage, Type: pipelinev1beta1.ParamTypeString},
 			{Name: PipelineIgnoredArtifacts, Type: pipelinev1beta1.ParamTypeString},
-			{Name: PipelineCacheUrl, Type: pipelinev1beta1.ParamTypeString, Default: &pipelinev1beta1.ArrayOrString{Type: pipelinev1beta1.ParamTypeString, StringVal: "http://jvm-build-workspace-artifact-cache." + namespace + ".svc.cluster.local/v1/cache/default/" + strconv.FormatInt(commitTime, 10)}},
-			{Name: "NAMESPACE", Type: pipelinev1beta1.ParamTypeString, Default: &pipelinev1beta1.ArrayOrString{Type: pipelinev1beta1.ParamTypeString, StringVal: namespace}},
+			{Name: PipelineCacheUrl, Type: pipelinev1beta1.ParamTypeString, Default: &pipelinev1beta1.ArrayOrString{Type: pipelinev1beta1.ParamTypeString, StringVal: "http://jvm-build-workspace-artifact-cache.$(context.pipelineRun.namespace).svc.cluster.local/v1/cache/default/" + strconv.FormatInt(commitTime, 10)}},
```

Comment:
ok yeah looking at the "don't hard code the namespace" commit, same comment - as was explained to me, though I have not even attempted to set up a KCP env with muliple compute clustes, it is entirely possible that the artiffact cache that a KCP client may create in the same logical workspace/namespace as this pipeline run may land on a different cluster

and as such, the k8s local service HTTP URL convention will fail, as the DNS / route for that does not span compute clusters

Reply:
but given that, without having ever implemented controller access to an ingress route before, I suspect these changes help us go in that direction.